L1Trigger/TrackFindingTracklet/interface/FitTrack.h

```diff
@@ -0,0 +1,1177 @@
+
+        // Get seed stubs first
+        trackstublist.push_back(std::make_pair(tracklet->innerFPGAStub(), tracklet->innerStub()));
+        if (tracklet->getISeed() >= 8)
+          trackstublist.push_back(std::make_pair(tracklet->middleFPGAStub(), tracklet->middleStub()));
+        trackstublist.push_back(std::make_pair(tracklet->outerFPGAStub(), tracklet->outerStub()));
+
+        // Now get ALL matches (can have multiple per layer)
+        for (unsigned int i = 0; i < fullmatch1_.size(); i++) {
+          for (unsigned int j = 0; j < fullmatch1_[i]->nMatches(); j++) {
+            if (fullmatch1_[i]->getFPGATracklet(j)->TCID() == tracklet->TCID()) {
+              trackstublist.push_back(fullmatch1_[i]->getMatch(j).second);
+            }
+          }
+        }
+
+        for (unsigned int i = 0; i < fullmatch2_.size(); i++) {
+          for (unsigned int j = 0; j < fullmatch2_[i]->nMatches(); j++) {
+            if (fullmatch2_[i]->getFPGATracklet(j)->TCID() == tracklet->TCID()) {
+              trackstublist.push_back(fullmatch2_[i]->getMatch(j).second);
+            }
+          }
+        }
+
+        for (unsigned int i = 0; i < fullmatch3_.size(); i++) {
+          for (unsigned int j = 0; j < fullmatch3_[i]->nMatches(); j++) {
+            if (fullmatch3_[i]->getFPGATracklet(j)->TCID() == tracklet->TCID()) {
+              trackstublist.push_back(fullmatch3_[i]->getMatch(j).second);
+            }
+          }
+        }
+
+        for (unsigned int i = 0; i < fullmatch4_.size(); i++) {
+          for (unsigned int j = 0; j < fullmatch4_[i]->nMatches(); j++) {
+            if (fullmatch4_[i]->getFPGATracklet(j)->TCID() == tracklet->TCID()) {
+              trackstublist.push_back(fullmatch4_[i]->getMatch(j).second);
+            }
+          }
+        }
+
+        // For merge removal, loop through the resulting list of stubs to calculate their stubids
+        if (RemovalType == "merge") {
+          for (std::vector<std::pair<Stub*, L1TStub*>>::iterator it = trackstublist.begin(); it != trackstublist.end();
+               it++) {
+            int layer = it->first->layer().value() + 1;  // Assume layer (1-6) stub first
+            if (it->first->layer().value() < 0) {        // if disk stub, though...
+              layer = it->first->disk().value() +
+                      10 * it->first->disk().value() / abs(it->first->disk().value());  //disk = +/- 11-15
+            }
+            stubidslist.push_back(
+                std::make_pair(layer, (it->first->phiregion().value() << 7) + it->first->stubindex().value()));
+          }
+
+          // And that's all we need! The rest is just for fitting, which is duplicated in PurgeDuplicate out of necessity
+          return;
+        }
+
+        HybridFit hybridFitter(iSector_, extended_, nHelixPar_);
+        hybridFitter.Fit(tracklet, trackstublist);
+        return;
+      }
+#endif
+
+      static TrackDerTable derTable;
```

Comment:
don't use non-`const` `static` variables

Reply:
fixed in the updated version